packages/headless/src/lib/headless.service.ts

```diff
@@ -0,0 +1,531 @@
+    off: (event: string) => void;
+    disconnect: () => void;
+  } | null = null;
+  private session: ISession | null = null;
+  private sessionQueryOptions: QueryObserverOptions<ISession, unknown> = {
+    queryKey: SESSION_QUERY_KEY,
+    cacheTime: Infinity,
+    staleTime: Infinity,
+    queryFn: () =>
+      this.api.initializeSession(
+        this.options.applicationIdentifier,
+        this.options.subscriberId,
+        this.options.subscriberHash
+      ),
+  };
+  private organizationQueryOptions: QueryObserverOptions<
+    IOrganizationEntity,
+    unknown
+  > = {
+    queryKey: ORGANIZATION_QUERY_KEY,
+    cacheTime: Infinity,
+    staleTime: Infinity,
+    queryFn: () => this.api.getOrganization(),
+  };
+  private unseenCountQueryOptions: QueryObserverOptions<
+    { count: number },
+    unknown
+  > = {
+    queryKey: UNSEEN_COUNT_QUERY_KEY,
+    queryFn: () => this.api.getUnseenCount(),
+  };
+  private userPreferencesQueryOptions: QueryObserverOptions<
+    IUserPreferenceSettings[],
+    unknown
+  > = {
+    queryKey: USER_PREFERENCES_QUERY_KEY,
+    queryFn: () => this.api.getUserPreference(),
+  };
+
+  constructor(private options: IHeadlessServiceOptions) {
+    const backendUrl = options.backendUrl ?? 'https://api.novu.co';
+    const token = getToken();
+    this.api = new ApiService(backendUrl);
+    this.applyToken(token);
+
+    this.queryClient = new QueryClient({
+      defaultOptions: {
+        queries: {
+          refetchOnWindowFocus: false,
+          retry: options?.config?.retry ?? 3,
+          ...(options?.config?.retryDelay
+            ? { retryDelay: options?.config?.retryDelay }
+            : {}),
+        },
+        mutations: {
+          retry: options?.config?.retry ?? 3,
+          ...(options?.config?.retryDelay
+            ? { retryDelay: options?.config?.retryDelay }
+            : {}),
+        },
+      },
+    });
+    this.queryService = new QueryService(this.queryClient);
+  }
+
+  private assertSessionInitialized() {
+    if (!this.session) {
+      throw new Error(SESSION_NOT_INITIALIZED_ERROR);
+    }
+  }
+
+  private applyToken(newToken: string | null) {
+    if (newToken) {
+      isBrowser &&
+        localStorage.setItem(NOTIFICATION_CENTER_TOKEN_KEY, newToken);
+      this.api.setAuthorizationToken(newToken);
+    } else {
+      isBrowser && localStorage.removeItem(NOTIFICATION_CENTER_TOKEN_KEY);
+      this.api.disposeAuthorizationToken();
+    }
+  }
+
+  private initializeSocket(token: string | null) {
+    const socketUrl = this.options.socketUrl ?? 'https://ws.novu.co';
+
+    if (this.socket) {
+      this.socket.disconnect();
+    }
+
+    if (token) {
+      this.socket = io(socketUrl, {
+        reconnectionDelayMax: 10000,
+        transports: ['websocket'],
+        query: {
+          token: `${token}`,
+        },
+      });
+
+      this.socket.on('connect_error', (e) => {
+        // eslint-disable-next-line no-console
+        console.error(e);
```

Comment:
As this will render in the dev tools console should we only console.error the error mesage and not the full trace?

Reply:
Makes sense. I've just checked and the error that is passed here has the message, I'll add it instead. Thanks!